superset/views/geocoding.py

```diff
@@ -0,0 +1,327 @@
+            )
+
+    def _geocode(self, data, dev=False):
+        """
+        Internal method which starts the geocoding
+        :param data: the data to be geocoded as a list of tuples
+        :param dev: Whether to Mock the geocoding process for testing purposes
+        :return: a list of tuples containing the data and the corresponding long, lat values
+        """
+        # TODO replace mock-method with mock-geocoder
+        if dev:
+            return self.coder.geocode("", data)
+        else:
+            return self.coder.geocode("MapTiler", data)
+
+    def _add_lat_lon_columns(self, table_name: str, lat_column: str, lon_column: str):
+        """
+        Add new longitued and latitude columns to table
+        :param table_name: The table name of table to insert columns
+        :param lat_column: The name of latitude column
+        :param lon_column: The name of longitude column
+        :raise SqlAddColumnException: When add column-SQL went wrong
+        """
+        connection = db.engine.connect()
+        transaction = connection.begin()
+        try:
+            self._add_column(connection, table_name, lat_column, Float())
+            self._add_column(connection, table_name, lon_column, Float())
+            transaction.commit()
+        except Exception:
+            transaction.rollback()
+            raise SqlAddColumnException(
+                "An error occured while creating new columns for latitude and longitude"
+            )
+
+    def _add_column(
+        self,
+        connection: Connection,
+        table_name: str,
+        column_name: str,
+        column_type: str,
+    ):
+        """
+        Add new column to table
+        :param connection: The connection to work with
+        :param table_name: The name of table to add a new column
+        :param column_name: The name of latitude column
+        :param column_type: The name of longitude column
+        """
+        column = Column(column_name, column_type)
+        name = column.compile(column_name, dialect=db.engine.dialect)
+        type = column.type.compile(db.engine.dialect)
+        sql = text("ALTER TABLE %s ADD %s %s" % (table_name, name, type))
+        connection.execute(sql)
+
+    def _insert_geocoded_data(
+        self,
+        table_name: str,
+        lat_column: str,
+        lon_column: str,
+        geo_columns: list,
+        data: list,
+    ):
+        """
+        Insert geocoded coordinates in table
+        :param table_name: The name of table to insert
+        :param lat_column: The name of latitude column
+        :param lon_column: The name of longitude column
+        :param geo_columns: The list of selected geographical column names
+        :param data: row with geographical data and geocoded coordinates
+        :raise SqlUpdateException: When Update of given columns with given data went wrong
+        """
+        where_clause = "='%s' AND ".join(filter(None, geo_columns)) + "='%s'"
```

Comment:
Does this work? this would lead to an sql-clause looking like "=%s  AND street,city,country = %s" right?

Reply:
yes, because the join builds the column names in the list with the previous string, but only "number in list -1"

Reply:
Did not realise the first part will be used as "char" inbetween the strings inserted, nice construct